### ISSUE TITLE
No method matching pointerforcomparison(::FastRange)

### ISSUE THREAD

**Issue by grahamas** [chriselrod/LoopVectorization.jl#203]
When indexing ranges, `@avx` fails, but when the same ranges are `collect`ed, `@avx` works just fine.

```julia
using LoopVectorization
function lv_test(us::AbstractVector{T}, vs::AbstractVector{T}) where T
    field_vals = Array{T,2}(undef, length(us), length(vs))
    @avx for u_idx ∈ axes(us,1), v_idx ∈ axes(vs,1)
        field_vals[u_idx, v_idx] = us[u_idx] * vs[v_idx]
    end
    field_vals
end
lv_test(0.1:0.1:10., 0.1:0.1:10.)     # produces below error
lv_test(0.1:0.1:10. |> collect, 0.1:0.1:10. |> collect)    # no error
```

```
ERROR: MethodError: no method matching pointerforcomparison(::VectorizationBase.FastRange{Float64, Float64, Float64, ArrayInterface.StaticInt{0}})
Closest candidates are:
  pointerforcomparison(::VectorizationBase.StridedBitPointer{N, C, B, R, X, O} where O where X where R where B where C) where N at /home/graham/.julia/packages/VectorizationBase/9kbnB/src/strided_pointers/stridedpointers.jl:240
  pointerforcomparison(::VectorizationBase.OffsetPrecalc) at /home/graham/.julia/packages/VectorizationBase/9kbnB/src/strided_pointers/cse_stridemultiples.jl:7
  pointerforcomparison(::VectorizationBase.AbstractStridedPointer) at /home/graham/.julia/packages/VectorizationBase/9kbnB/src/strided_pointers/cse_stridemultiples.jl:10
  ...
Stacktrace:
 [1] macro expansion
   @ ~/.julia/packages/LoopVectorization/8MeJu/src/reconstruct_loopset.jl:526 [inlined]
 [2] _avx_!
   @ ~/.julia/packages/LoopVectorization/8MeJu/src/reconstruct_loopset.jl:526 [inlined]
 [3] lv_test(us::StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}, vs::StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}})
   @ Main ./REPL[2]:4
 [4] top-level scope
   @ REPL[3]:1
```

version edit:
```
julia> versioninfo()
Julia Version 1.6.0-beta1
Commit b84990e1ac (2021-01-08 12:42 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-10710U CPU @ 1.10GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.0 (ORCJIT, skylake)
Environment:
  JULIA_EDITOR = "/snap/code/55/usr/share/code/code"
  JULIA_NUM_THREADS = 
...
[bdcacae8] LoopVectorization v0.11.2
```

**Comment by chriselrod** [comment]
Thanks for the issue.
It'll [be fixed on VectorizationBase 0.18.11](https://github.com/chriselrod/VectorizationBase.jl/commit/77564fa3347ad21a0fe3ff0a36cbec3b709a8699), which [will be released shortly](https://github.com/JuliaRegistries/General/pull/30768)

**Comment by grahamas** [comment]
Awesome, thanks!

Re-testing in an environment with latest VectorizationBase works perfectly. Thanks for such a fast response (and thanks for the package! -- it sped up my code 30x yesterday!)
